src/Common/src/Interop/Windows/kernel32/Interop.GetComputerName.cs

```diff
@@ -9,18 +9,19 @@ internal partial class Interop
     internal partial class Kernel32
     {
         [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, EntryPoint = "GetComputerNameW")]
-        private extern static int GetComputerName(char[] lpBuffer, ref uint nSize);
+        private static extern unsafe int GetComputerName(char* lpBuffer, ref uint nSize);
 
-        private const int MacMachineNameLength = 256;
+        // maximum length of the NETBIOS name (not including NULL)
+        private const int MAX_COMPUTERNAME_LENGTH = 15;
 
-        internal static string GetComputerName()
+        internal static unsafe string GetComputerName()
         {
-            char[] buffer = new char[MacMachineNameLength];
-            uint length = (uint)buffer.Length;
+            char* buffer = stackalloc char[MAX_COMPUTERNAME_LENGTH + 1];
```

Comment:
I believe that'd require a cast, e.g. `char[(int)length]`, but ok. I can change tomorrow.